### DISCUSSION TITLE
LDM Notes for October 4th, 2023

### DISCUSSION THREAD

**Discussion by 333fred** [dotnet/csharplang#7575] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2023/LDM-2023-10-04.md

## Agenda

- Trimming and AOT

**Comment by jnm2** [comment]
(second to last page of https://github.com/dotnet/csharplang/blob/main/meetings/2023/LDM-2023-10-04%20Intro%20to%20Trimming%20and%20AOT.pdf:)

> ## C# lessons
>
> Source generator per library doesn't scale

I'm curious to hear more about this.

**Reply by agocke** [reply]
Scale in terms of developer cost, not performance. If things as simple as EventSource need to write a source generator, huge numbers of libraries will need source generators. It's too expensive to write a source generator for every small serialization library in the ecosystem.

**Reply by jnm2** [reply]
Can you elaborate on the feedback or rationale behind why that would be too expensive for every small serialization library to do? I have some guesses, but I'd love to follow the thought process.

**Reply by agocke** [reply]
Writing a source generator is just complex. It's an advanced user scenario. Pushing it to every serialization format isn't efficient.

**Comment by huoyaoyuan** [comment]
> Any further addition of language features that doesn't have real runtime representation makes it harder to work with from a cross-language perspective

Highlights.

To talk further, there are more features that requires both language and runtime work. For example, improvements to delegate and enum types, which are declared type from compilers.
